src/Helpers/babylon.particleHelper.ts

```diff
@@ -0,0 +1,246 @@
+            
+            return new Promise((resolve, reject) => {
+                if (scene) {
+                    this._scene = scene;
+                } else {
+                    return reject("A particle system need a scene.");
+                }
+
+                if (gpu && !GPUParticleSystem.IsSupported) {
+                    return reject("Particle system with GPU is not supported.");
+                }
+
+                this._emitter = emitter;
+
+                Tools.LoadFile(`${this._baseAssetsUrl}/systems/${type}.json`, (data, response) => {
+                    const newData = JSON.parse(data.toString()) as IParticleSystemData;
+                    return resolve(this._createSystem(newData));
+                }, undefined, undefined, undefined, (req, exception) => {
+                    return reject(`An error occured while the creation of your particle system. Check if your type '${type}' exists.`);
+                });
+
+            });
+        }
+
+        private static _createSystem(data: IParticleSystemData): ParticleSystem {
+            // Create a particle system
+            const system = new ParticleSystem(data.type, data.capacity, this._scene);
+            // Texture of each particle
+            system.particleTexture = new Texture(`${this._baseAssetsUrl}/textures/${data.textureFile}`, this._scene);
+            // Where the particles come from
+            system.emitter = this._emitter; // the starting object, the emitter
+
+            // Colors of all particles
+            system.color1 = new Color4(data.color1.r, data.color1.g, data.color1.b, data.color1.a);
+            system.color2 = new Color4(data.color2.r, data.color2.g, data.color2.b, data.color2.a);
+            system.colorDead = new Color4(data.colorDead.r, data.colorDead.g, data.colorDead.b, data.colorDead.a);
+
+            // Size of each particle (random between...
+            system.minSize = data.minSize;
+            system.maxSize = data.maxSize;
+
+            // Life time of each particle (random between...
+            system.minLifeTime = data.minLifeTime;
+            system.maxLifeTime = data.maxLifeTime;
+
+            // Emission rate
+            system.emitRate = data.emitRate;
+
+            // Blend mode : BLENDMODE_ONEONE, or BLENDMODE_STANDARD
+            system.blendMode = data.blendMode;
+
+            // Set the gravity of all particles
+            system.gravity = new Vector3(data.gravity.x, data.gravity.y, data.gravity.z);
+
+            // Angular speed, in radians
+            system.minAngularSpeed = data.minAngularSpeed;
+            system.maxAngularSpeed = data.maxAngularSpeed;
+
+            // Speed
+            system.minEmitPower = data.minEmitPower;
+            system.maxEmitPower = data.maxEmitPower;
+            system.updateSpeed = data.updateSpeed;
+
+            switch (data.emitterType) {
```

Comment:
no need for an enum here. Plain string is fine

Reply:
good like that?

Reply:
yes!

Reply:
ahah great